### ISSUE TITLE
Method `$.shouldHave(text)` should verify the whole text (not a substring)

### ISSUE THREAD

**Issue by asolntsev** [selenide/selenide#1780]
## The problem
Current behaviour of `$.shouldHave(text())` is misleading. People might think it verifies the whole text, but it actually verifies only a substring. 

## For example
The test: `$.shouldHave(text("error"));` may be green even if html has text `<div>NO errors</div>` which has the opposite meaning. This behaviour can cause false positive tests. 

## Historical reasons
Many years ago I decided to check a substring in order to ignore spaces, tabs, newlines etc. Now it seems it was not really needed because the condition `text` already implements this logic inside. 

## For backward compatibility
I suggest to 
1. change the default behaviour of `$.shouldHave(text())` to verify the whole text
2. introduce one more setting like `Configuration.textCheck = FULL_TEXT | PARTIAL_TEXT;` which allows users to restore the previous behaviour in case if they get a lot of failing tests.

## Tell us about your environment
* **Selenide Version**: 6.4.0

**Comment by plagov** [comment]
In the beginning, one might indeed think that `text()` would check the whole exact text of the element.

Although, the option with a configuration is backwards compatible, having yet another configuration is not the best solution, IMO.

If the decision would be to switch the `text()` to a full-text match, then we definitely need something like a `partialText()` condition for a substring matching.

**Comment by asolntsev** [comment]
@plagov Yes, PR https://github.com/selenide/selenide/pull/1783 introduces `partialText()`.

**Comment by cocorossello** [comment]
I would go with partialText() and exactText() and deprecate text(). Maybe those should also have an optional flag to make it case sensitive?

Changing text() to check the exact text is a big breaking change, in my opinion.

**Comment by asolntsev** [comment]
@cocorossello Sure, it's a breaking change. I suggest to make it configurable: 
```java
  Configuration.textCheck = PARTIAL_TEXT;   // current default
  Configuration.textCheck = FULL_TEXT;         // the "new way". 
```

We could make `FULL_TEXT` default in next major release 7.0.0

**Comment by plagov** [comment]
@asolntsev do you have any plans or thoughts on the `exactText()` assertion once the `text()` one will check for the full text by default after some time? Will it be deprecated in favour of the `text()` or left as an alias/synonym?

**Comment by asolntsev** [comment]
@plagov Yes, when `Configuration.textCheck = FULL_TEXT` then `exactText()` is synonym for just `text()`. 
May be we can mark them as deprecated in some future versions of Selenide. We'll see.

**Comment by asolntsev** [comment]
Hm... Good point!

But no, FULL-TEXT was not made the default in Selenide 7.0.0

I am still afraid that most users hasn't enabled this flag, and this change would trigger too big hassle.. What do you think?

**Comment by asolntsev** [comment]
To be honest, I forgot about FULL_TEXT when released 7.0.0.
I looked at all "deprecated" words in the project, but FULL_TEXT was not "deprecated". :)
